stdlib/public/Concurrency/Task.swift

```diff
@@ -495,6 +495,27 @@ extension Task {
   public static func yield() async {
     fatalError("\(#function) not implemented yet.")
   }
+
+  /// Suspends the current task for _at least_ the given duration
+  /// in nanoseconds.
+  ///
+  /// This function does _not_ block the underlying thread.
+  public static func sleep(_ duration: UInt64) async {
+
+    // Set up the job flags for a new task.
+    var flags = JobFlags()
+    flags.kind = .task
+    flags.priority = .default
+    flags.isFuture = true
+
+    // Create the asynchronous task future.
+    let (task, _) = Builtin.createAsyncTaskFuture(flags.bits, nil, {})
+
+    // Enqueue the resulting job.
+    _enqueueJobGlobalWithDelay(duration, Builtin.convertTaskToJob(task))
+
+    let _ = await Handle<Void, Never>(task).get()
+  }
```

Comment:
Hmm, I wonder if implementing this without a detached implicitly created task might be better?

It'd need much more boilerplate to introduce a new "awaitable" function like I did for `swift_task_group_wait_next` for groups... It'd save some allocations I guess, though not sure if worth it since this is a meh sleep anyway.

What do @DougGregor @jckarter think?